traitlets/traitlets.py

```diff
@@ -2479,7 +2479,16 @@ def instance_init(self, obj):
 
     def from_string(self, s):
         """Load value from a single string"""
-        return self.from_string_list([s])
+        if not isinstance(s, str):
```

Comment:
It seems like calling `from_string_list` as the *default* for containers makes sense to me, since I would expect most containers to only define `from_string_list`. Does it not? Custom traits can override this to give different single-string behavior, but it seems like the right thing to do for the traits we have. What do you think?

Reply:
It get weird on edge case; like for example "from_string('{}')", will call "from_string_list(['{}'])", which does the wrong thing.

Reply:
Right.